### ISSUE TITLE
Investigate inconsistent ssz deserialize

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#2648]
**Describe the bug**

Given this block 32768 in oonoonba network 
[signed_block_32768.zip](https://github.com/ChainSafe/lodestar/files/6613454/signed_block_32768.zip)

```ts
const signedBlock = ssz.altair.SignedBeaconBlock.createTreeBackedFromBytes(signedBlockBuf);
console.log("@@@ loaded signed block successfully at slot", signedBlock.message.slot, "root", toHexString(ssz.altair.BeaconBlock.hashTreeRoot(signedBlock.message)));
```
gives us `@@@ loaded signed block successfully at slot 32768 root 0xc1ad6fc70d6c83cba35288fc58441325bbb985b16b407bdff2857ba6a30fafd0` which is correct (since we can process that block)

while
```ts
const signedBlock = ssz.altair.SignedBeaconBlock.deserialize(signedBlockBuf);
console.log("@@@ loaded signed block successfully at slot", signedBlock.message.slot, "root", toHexString(ssz.altair.BeaconBlock.hashTreeRoot(signedBlock.message)));
```
gives us `@@@ loaded signed block successfully at slot 32768 root 0x9b1482b3be4086d530feb8954f6b5a3a975381c26d2a97482f51db974b5fd59a`

this is incorrect block root and we got `Invalid block signature` processing that block

**Expected behavior**

+ `ssz.altair.SignedBeaconBlock.deserialize()` should be the same to `ssz.altair.SignedBeaconBlock.createTreeBackedFromBytes()`
+ Need to investigate what caused the issue at ssz level

**Comment by twoeths** [comment]
+ there is no issue with `equal`, it's just issue with `hashTreeRoot`
+ `hashTreeRoot(signedBlockStruct.message.body.syncAggregate.syncCommitteeBits)` turns out to be incorrect

Note that `BitVectorType` is a new type in altair, that's why we haven't seen the issue during phase0 sync.

**Comment by dapplion** [comment]
> * there is no issue with `equal`, it's just issue with `hashTreeRoot`
> * `hashTreeRoot(signedBlockStruct.message.body.syncAggregate.syncCommitteeBits)` turns out to be incorrect
> 
> Note that `BitVectorType` is a new type in altair, that's why we haven't seen the issue during phase0 sync.

Should we add unit tests for that type in the ssz repo?

**Comment by twoeths** [comment]
we haven't implemented ssz spec tests, that should scan these ssz issues if we test both struct type and tree backed type

**Comment by dapplion** [comment]
> we haven't implemented ssz spec tests, that should scan these ssz issues if we test both struct type and tree backed type

Should we prioritize implementing those then?

**Comment by twoeths** [comment]
Fixed in ssz 0.8.8
